src/managers/TileManager.ts

```diff
@@ -47,6 +48,25 @@ class TileManager {
         );
     }
 
+    public getStartingTiles(): Tile[] {
+        const session: Nullable<SessionDTO> = GeneralManager.getSession();
+        assert(session);
+        const tiles: Tile[] = [];
+        const random: seedrandom.PRNG = seedrandom(session.seed);
+        const veins: int[] = seededShuffle([9, 10, 10], session.seed); // 9 is gold, 10 are coal veins
+
+        for (const item of preplacedTiles) {
+            const tile: Tile = new Tile(random, item.type || veins.pop()!);
+
+            tile.coordinateX = item.coordinateX;
+            tile.coordinateY = item.coordinateY;
+            tile.rotation = item.rotation;
+            tile.state = TileState.Placed;
```

Comment:
please dont assign the properties outside... use .apply(state, dto) and fake a dto with the coordinates and rotation